cmake/sysbuild/partition_manager.cmake

```diff
@@ -126,106 +199,153 @@ endif()
-  set(otp_start_addr "0xff8108")
-  set(otp_size 756) # 189 * 4
-elseif (DEFINED CONFIG_SOC_NRF5340_CPUAPP)
-  # See nRF5340 Product Specification, chapter Application Core -> ... "UICR"
-  set(otp_start_addr "0xff8100")
-  set(otp_size 764)  # 191 * 4
-endif()
+set(DOMAIN_APP_MAIN ${app_name})
+foreach(d MAIN ${PM_DOMAINS})
+  # CPUNET
+  set(image_name ${DOMAIN_APP_${d}})
+  if(${d} STREQUAL "MAIN")
+    set(d)
+  endif()
+  sysbuild_get(${image_name}_CONFIG_PM_SRAM_SIZE IMAGE ${image_name} VAR CONFIG_PM_SRAM_SIZE KCONFIG)
+  sysbuild_get(${image_name}_CONFIG_PM_SRAM_BASE IMAGE ${image_name} VAR CONFIG_PM_SRAM_BASE KCONFIG)
+
+  sysbuild_get(${image_name}_CONFIG_SOC_NRF9160 IMAGE ${image_name} VAR CONFIG_SOC_NRF9160 KCONFIG)
+  sysbuild_get(${image_name}_CONFIG_SOC_NRF5340_CPUAPP IMAGE ${image_name} VAR CONFIG_SOC_NRF5340_CPUAPP KCONFIG)
+
+  if (DEFINED ${image_name}_CONFIG_SOC_NRF9160)
+    # See nRF9160 Product Specification, chapter "UICR"
+    set(otp_start_addr "0xff8108")
+    set(otp_size 756) # 189 * 4
+  elseif (DEFINED ${image_name}_CONFIG_SOC_NRF5340_CPUAPP)
+    # See nRF5340 Product Specification, chapter Application Core -> ... "UICR"
+    set(otp_start_addr "0xff8100")
+    set(otp_size 764)  # 191 * 4
+  endif()
+
+  add_region(
+    NAME sram_primary
+    SIZE ${${image_name}_CONFIG_PM_SRAM_SIZE}
+    BASE ${${image_name}_CONFIG_PM_SRAM_BASE}
+    PLACEMENT complex
+    DYNAMIC_PARTITION sram_primary
+    DOMAIN ${d}
+    )
 
-sysbuild_get(${app_name}_CONFIG_PM_SRAM_SIZE IMAGE ${app_name} VAR CONFIG_PM_SRAM_SIZE KCONFIG)
-sysbuild_get(${app_name}_CONFIG_PM_SRAM_BASE IMAGE ${app_name} VAR CONFIG_PM_SRAM_BASE KCONFIG)
+  sysbuild_get(${image_name}_CONFIG_FLASH_SIZE IMAGE ${image_name} VAR CONFIG_FLASH_SIZE KCONFIG)
+  math(EXPR flash_size "${${image_name}_CONFIG_FLASH_SIZE} * 1024" OUTPUT_FORMAT HEXADECIMAL)
 
-add_region(
-  NAME sram_primary
-  SIZE ${${app_name}_CONFIG_PM_SRAM_SIZE}
-  BASE ${${app_name}_CONFIG_PM_SRAM_BASE}
-  PLACEMENT complex
-  DYNAMIC_PARTITION sram_primary
-  )
+  if (${image_name}_CONFIG_SOC_NRF9160 OR ${image_name}_CONFIG_SOC_NRF5340_CPUAPP)
+    add_region(
+      NAME otp
+      SIZE ${otp_size}
+      BASE ${otp_start_addr}
+      PLACEMENT start_to_end
+      DOMAIN ${d}
+      )
+  endif()
+  sysbuild_get(${image_name}_CONFIG_FLASH_BASE_ADDRESS IMAGE ${image_name} VAR CONFIG_FLASH_BASE_ADDRESS KCONFIG)
+  add_region(
+    NAME flash_primary
+    SIZE ${flash_size}
+    BASE ${${image_name}_CONFIG_FLASH_BASE_ADDRESS}
+    PLACEMENT complex
+    DEVICE flash_controller
+    DEFAULT_DRIVER_KCONFIG CONFIG_SOC_FLASH_NRF
+    DOMAIN ${d}
+    )
 
-sysbuild_get(${app_name}_CONFIG_FLASH_SIZE IMAGE ${app_name} VAR CONFIG_FLASH_SIZE KCONFIG)
-math(EXPR flash_size "${${app_name}_CONFIG_FLASH_SIZE} * 1024" OUTPUT_FORMAT HEXADECIMAL)
 
-if (CONFIG_SOC_NRF9160 OR CONFIG_SOC_NRF5340_CPUAPP)
+endforeach()
+
+sysbuild_get(ext_flash_enabled IMAGE ${app_name} VAR CONFIG_PM_EXTERNAL_FLASH_ENABLED KCONFIG)
```

Comment:
Think things like this are global if I remember rightly, i.e. enable in main image and it gets set in child images. If so, might be worth moving to be a sysbuild Kconfig (and controlled) entry

Reply:
yep, see also this comment https://github.com/nrfconnect/sdk-nrf/pull/11015#discussion_r1222622302